packages/nimble-components/src/table-column/icon/tests/table-column-icon.spec.ts

```diff
@@ -0,0 +1,398 @@
+            expect(column.validity.missingKeyValue).toBeFalse();
+            expect(column.validity.invalidIconName).toBeFalse();
+        });
+
+        it('is valid with valid numeric key values', async () => {
+            ({ element, connect, disconnect, model } = await setup(
+                [
+                    { key: '0', text: 'alpha', icon: iconXmarkTag },
+                    { key: '1', text: 'alpha', icon: iconXmarkTag },
+                    { key: '1.01', text: 'alpha', icon: iconXmarkTag },
+                    { key: '-1.01', text: 'alpha', icon: iconXmarkTag },
+                    { key: '-1e3', text: 'alpha', icon: iconXmarkTag }
+                ],
+                'number'
+            ));
+            await connect();
+            await waitForUpdatesAsync();
+            expect(model.col1.checkValidity()).toBeTrue();
+            expect(
+                model.col1.validity.invalidMappingKeyValueForType
+            ).toBeFalse();
+        });
+
+        describe('is invalid with invalid boolean key values:', () => {
+            const dataTypeTests = [
+                { name: '(blank)', key: '' },
+                { name: 'FALSE', key: 'FALSE' },
+                { name: '0', key: 0 }
+            ];
+            const focused: string[] = [];
+            const disabled: string[] = [];
+            for (const test of dataTypeTests) {
+                const specType = getSpecTypeByNamedList(
+                    test,
+                    focused,
+                    disabled
+                );
+                // eslint-disable-next-line @typescript-eslint/no-loop-func
+                specType(` ${test.name}`, async () => {
+                    ({ element, connect, disconnect, model } = await setup(
+                        [
+                            {
+                                key: test.key,
+                                text: 'alpha',
+                                icon: iconXmarkTag
+                            }
+                        ],
+                        'boolean'
+                    ));
+                    await connect();
+                    await waitForUpdatesAsync();
+                    expect(model.col1.checkValidity()).toBeFalse();
+                    expect(
+                        model.col1.validity.invalidMappingKeyValueForType
+                    ).toBeTrue();
+                });
+            }
+        });
+
+        it('is invalid with invalid numeric key values', async () => {
+            ({ element, connect, disconnect, model } = await setup(
+                [{ key: 'a', text: 'alpha', icon: iconXmarkTag }],
+                'number'
+            ));
+            await connect();
+            await waitForUpdatesAsync();
+            expect(model.col1.checkValidity()).toBeFalse();
+            expect(
+                model.col1.validity.invalidMappingKeyValueForType
+            ).toBeTrue();
+        });
+
+        describe('invalid mappings', () => {
```

Comment:
Again another case where the describe block offers strange chain semantics of "TableColumnIcon validation invalid mappings is invalid with text mapping" instead of "TableColumnIcon validation invalid with text mapping".